### PR TITLE
CopyPromise switched to lutimes

### PR DESCRIPTION
**Pull request by elmpp**
**What's the problem this PR addresses?**
Was seeing errors whilst installing with a `file:/` descriptor.

Probably could produce a minimal repro but encountered this whilst defining a file: dependency to a workspace within an enveloping project. The cwd was a package inside the project but with an empty lockfile. 

Here is a [hastebin](https://hastebin.com/pafikobigo) showing local repro and fix confirmation

...

**How did you fix it?**
switched the use of `utime` with `lutime` as suggested [here](https://discordapp.com/channels/226791405589233664/654372321225605128/707888149786787851)
...

### CODE REVIEW

File: packages/yarnpkg-fslib/sources/algorithms/copyPromise.ts [hunk 1]
```diff
@@ -10,45 +10,46 @@ export type CopyOptions = {
...
+export type LUTimes<P extends Path> =
   Array<[P, Date, Date]>;
 
 export async function copyPromise<P1 extends Path, P2 extends Path>(destinationFs: FakeFS<P1>, destination: P1, sourceFs: FakeFS<P2>, source: P2, opts: CopyOptions) {
   const normalizedDestination = destinationFs.pathUtils.normalize(destination);
   const normalizedSource = sourceFs.pathUtils.normalize(source);
 
   const operations: Operations = [];
-  const utimes: UTimes<P1> = [];
+  const lutimes: LlUTimes<P1> = [];
```
Comment (arcanis) [inline packages/yarnpkg-fslib/sources/algorithms/copyPromise.ts:21]:
```suggestion
  const lutimes: LUTimes<P1> = [];
```

File: packages/yarnpkg-fslib/sources/algorithms/copyPromise.ts [hunk 2]
```diff
@@ -10,45 +10,46 @@ export type CopyOptions = {
...
 
   for (const operation of operations)
     await operation();
 
-  for (const [p, atime, mtime] of utimes) {
-    await destinationFs.utimesPromise(p, atime, mtime);
+  for (const [p, atime, mtime] of lutimes) {
+    // await destinationFs.utimesPromise(p, atime, mtime);
+    return await (typeof destinationFs.lutimesPromise === 'function') ? destinationFs.lutimesPromise!(p, atime, mtime) : destinationFs.utimesPromise!(p, atime, mtime)
   }
```
Comment (arcanis) [inline packages/yarnpkg-fslib/sources/algorithms/copyPromise.ts:33]:
```suggestion
  const updateTime = typeof destinationFs.lutimesPromise === 'function'
    ? destinationFs.lutimesPromise
    : destinationFs.utimesPromise;

  for (const [p, atime, mtime] of lutimes) {
    // await destinationFs.utimesPromise(p, atime, mtime);
    await updateTime!(p, atime, mtime)
  }
```

File: packages/yarnpkg-fslib/sources/algorithms/copyPromise.ts [hunk 3]
```diff
@@ -28,8 +28,8 @@ export async function copyPromise<P1 extends Path, P2 extends Path>(destinationF
     await operation();
 
   const updateTime = typeof destinationFs.lutimesPromise === 'function'
-    ? destinationFs.lutimesPromise
-    : destinationFs.utimesPromise;
+    ? destinationFs.lutimesPromise.bind(destinationFs)
```
Comment (elmpp) [inline packages/yarnpkg-fslib/sources/algorithms/copyPromise.ts:31]: thanks

**Comment by arcanis** [comment]
Here we go! Thanks!

**Pull request merged**